twill-common/src/main/java/org/apache/twill/filesystem/Location.java

```diff
@@ -207,6 +207,21 @@
   boolean mkdirs() throws IOException;
 
   /**
+   * Creates the directory named by this abstract pathname, including any necessary
+   * but nonexistent parent directories.
+   *
+   * @param permission A permission string. It has to be either a three digit or a nine character string.
+   *                   For the three digit string, it is similar to the UNIX permission numeric representation.
+   *                   The first digit is the permission for owner, second digit is the permission for group and
+   *                   the third digit is the permission for all.
+   *                   For the nine character string, it uses the format as specified by the
+   *                   {@link PosixFilePermissions#fromString(String)} method.
+   *
+   * @return true if and only if the renaming succeeded; false otherwise
```

Comment:
`@return true if and only if the renaming succeeded; false otherwise` => needs to change from renaming to mkdir